malaria-ios/Managers/MedicineStockManager.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+class MedicineStockManager: CoreDataContextManager {
+  
+  private let medicine: Medicine
+  
+  /// Init
+  init(medicine: Medicine) {
+    self.medicine = medicine
+    super.init(context: medicine.managedObjectContext!)
+  }
+  
+  ///
+  func updateStock(tookMedicine: Bool) {
+    Logger.Info("Updating stock.")
+    medicine.remainingMedicine += tookMedicine ? -1 : 1
+    CoreDataHelper.sharedInstance.saveContext(context)
+  }
+  
+  func addRegistry(date: NSDate, tookMedicine: Bool, modifyEntry: Bool = false) -> Bool {
+    
+    // Check if the user modifies a date, after the last refill, in the calendar
+    // if so, modify the entry and the current stock
+    // else, only modify the entry
+    let lastStockRefill = medicine.lastStockRefill ?? NSDate()
+    
+    if date.sameDayAs(lastStockRefill) || date > lastStockRefill {
+      
+      // Check if the user has enough pills to say that he took the pill
+      if !medicine.medicineStockManager.hasEnoughPills() && tookMedicine {
+        return false
+      }
+      
+      let registryAdded = medicine.registriesManager.addRegistry(date, tookMedicine: tookMedicine, modifyEntry: modifyEntry)
+      
+      // Only update stock if the entry was added successfully
+      if registryAdded.0 && registryAdded.noOtherEntryFound {
```

Comment:
What represents 0 ? I assume this is a tuple. Replace by proper identifiers to be clear what is happening.

e.g.
var person = (firstName: "John", lastName: "Smith")

person.firstName instead of person.0